### DISCUSSION TITLE
Individualize tgrid

### DISCUSSION THREAD

**Discussion by Mutaz94** [metrumresearchgroup/mrgsolve#825] [Q&A] [answered]
Hi @kylebaron-

Is there any way to individualize `tgrid`. For example, each subject will have different time point. I thought about `rnorm` but that will get all subject same sample. 

```
SAMPLE <- c(0.5, 1, 1.5,2,4,6)
tgrid= SAMPLE + rnorm(length(SAMPLE),0,1)
```
This above will sample all population with same times. 
Any ideas to make each subject different?

Thanks!

**Comment by kylebaron** [comment, marked as answer]
Hi @Mutaz94 - 

A couple of different ways to do it:

``` r
library(tidyverse)
library(mrgsolve)
#> 
#> Attaching package: 'mrgsolve'
#> The following object is masked from 'package:stats':
#> 
#>     filter
mod <- modlib("popex")
#> Building popex ...
#> done.


data <- as_data_set(
  ev(amt = 300, ii = 24, addl = 10, ID = 1:10, GROUP = 1)
)
```

# Option 1 - tgrid

``` r
set.seed(132)
dl <- map(data$ID, ~ sample(seq(0,24),5))

idata <- select(data, ID)

out <- mrgsim(
  mod, 
  data = data, 
  idata = idata, 
  deslist = dl, 
  descol = "ID"
) 

plot(out, DV~time, scales = "free", type = 'b')
```

![](https://i.imgur.com/SEhSRq1.png)

# Option 2 - build up data

``` r
set.seed(22233)
data2 <- 
  data %>% 
  select(-time) %>%
  mutate(
  time = map(ID, ~sample(seq(0,24),5))
) 

data2 <- 
  data2 %>%
  mutate(evid = 0, amt = 0, ii = 0)  %>%
  unnest(cols = "time") %>% 
  bind_rows(data) %>% 
  arrange(ID,time)

out <- mrgsim(mod, data2)
plot(out, DV ~ time, type = 'b')
```

![](https://i.imgur.com/JQCb2yD.png)

<sup>Created on 2021-05-05 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by Mutaz94** [comment]
THANK YOU!! thats exactly what I want. option 1 is the winner I guess.
